test/Promise.test.js

```diff
@@ -211,6 +211,74 @@ describe ('Promise', function () {
     });
   });
 
+  describe('finally', function () {
+    it('should call finally when resolved', function(done) {
+      var isFullfilled = false;
+      var finallyRan = false;
+      var p = new Promise(function(resolve, reject) {
+        resolve(1); 
+      }).then(function(value) {
+        isFullfilled = true; 
+      }).finally(function(value) {
+        assert.ok(isFullfilled, "should call finally after resolve");
+        assert.equal(value, undefined);
+        finallyRan = true;
+      }).then(function() {
+        assert.equal(finallyRan, true);
+      });
+
+      assert.strictEqual(p.resolved, true);
+      assert.strictEqual(p.rejected, false);
+      assert.strictEqual(p.pending, false);
+
+      done();
```

Comment:
I think the `done()` in this unit test should be _inside_ the last `.then()`, like:

```js
.then(function() {
  assert.equal(finallyRan, true);
  done();
});
```

It works right now, but in general, a promise runs async so the `done` would be called before the `assert.equal(finallyRan, true)`.

Reply:
Yes this makes sense. I was following the other tests in `Promise.test.js` I can refactor this to run in the promise chain.